agent/lib/kontena/workers/node_info_worker.rb

```diff
@@ -236,6 +253,58 @@ def calculate_container_time(container)
       0
     end
 
+    # @param [Array<Vmstat::Cpu>] prev_cpus
+    # @param [Array<Vmstat::Cpu>] current_cpu
+    # @return [Hash] { :num_cores, :system, :user, :idle }
+    def calculate_cpu_usage(prev_cpus, current_cpus)
+      result = {
+        num_cores: prev_cpus.size,
+        system: 0.0,
+        user: 0.0,
+        idle: 0.0
+      }
+
+      prev_cpus.zip(current_cpus).map { |prev_cpu, current_cpu|
+        system_ticks = current_cpu.system - prev_cpu.system
+        user_ticks = current_cpu.user - prev_cpu.user
+        idle_ticks = current_cpu.idle - prev_cpu.idle
+
+        total_ticks = (system_ticks + user_ticks + idle_ticks).to_f
+
+        {
+          system: (system_ticks / total_ticks) * 100.0,
+          user: (user_ticks / total_ticks) * 100.0,
+          idle: (idle_ticks / total_ticks) * 100.0
+        }
+      }.inject(result) { |memo, cpu_core|
+        memo[:system] += cpu_core[:system]
+        memo[:user] += cpu_core[:user]
+        memo[:idle] += cpu_core[:idle]
+        memo
+      }
+    end
+
+    # @param [Vmstat::NewtworkInterface] previous_iface
+    # @param [Vmstat::NetworkInterface] current_iface
+    # @param [Number] interval
+    # @return [Hash]
+    def calculate_network_traffic(previous_iface, current_iface, interval)
+      in_bytes_per_second = (current_iface.in_bytes - previous_iface.in_bytes) / interval
+      out_bytes_per_second = (current_iface.out_bytes - previous_iface.out_bytes) / interval
+
+      {
+        interface_name: current_iface.name,
+        in_bytes_per_second: in_bytes_per_second,
+        out_bytes_per_second: out_bytes_per_second
+      }
+    end
+
+    def get_network_interface
+      Vmstat.network_interfaces.select { |x| x.ethernet? and x.in_bytes > 0 and x.out_bytes > 0 }
```

Comment:
@SpComb @daniel-bytes if `x.ethernet?` returns only virtualized ethernet interfaces then we should probably gather stats from those and store them separately in the master database.